src/snapred/backend/data/GroceryService.py

```diff
@@ -270,20 +274,24 @@ def _fetchInstrumentDonor(self, runId: str, useLiteMode: bool) -> WorkspaceName:
                 loadEmptyInstrument.execute()
 
                 # Initialize the instrument parameters
-                # (Reserved IDs will use the unmodified instrument.)
-                if runId != GroceryListItem.RESERVED_NATIVE_RUNID and runId != GroceryListItem.RESERVED_LITE_RUNID:
-                    self._updateInstrumentParameters(runId, wsName)
+                # (Reserved run-numbers will use the unmodified instrument.)
+                if (
+                    runNumber != GroceryListItem.RESERVED_NATIVE_RUNNUMBER
+                    and runNumber != GroceryListItem.RESERVED_LITE_RUNNUMBER
+                ):
+                    detectorState: DetectorState = self._getDetectorState(runNumber)
+                    self.updateInstrumentParameters(wsName, detectorState)
```

Comment:
Just checking 90% of this file is just a changing `runid` -> `runNumber`, plus the code here